smartsim/_core/control/controller.py

```diff
@@ -346,6 +346,35 @@ def init_launcher(self, launcher: str) -> None:
         else:
             raise TypeError("Must provide a 'launcher' argument")
 
+    @staticmethod
+    def symlink_output_files(
+        job_step: Step, entity: t.Union[SmartSimEntity, EntitySequence[SmartSimEntity]]
+    ) -> None:
+        """Symlink output files under .smartsim to the entity output files
+
+        :param job_step: Job step instance
+        :type job_step: Step
+        :param entity: Entity instance
+        :type entity: SmartSimEntity | EntitySequence[SmartSimEntity]
+        """
+        historical_out, historical_err = map(pathlib.Path, job_step.get_output_files())
+        entity_out = pathlib.Path(entity.path) / f"{entity.name}.out"
+        entity_err = pathlib.Path(entity.path) / f"{entity.name}.err"
+
+        # check if there is already a link to a previous run
+        if entity_out.is_symlink() or entity_err.is_symlink():
+            entity_out.unlink()
+            entity_err.unlink()
+
+        try:
+            entity_out.symlink_to(historical_out)
```

Comment:
What guarantees that `entity_out` has already been created? Especially for a batch step (which could start after some time), do we create the files in advance? Or will they just be "broken" symlinks until the real files appear?

Reply:
`entity_out` is really just a reference that points to `historical_out`, so we don't need to create actual files on the filesystem for `entity_out` or `entity_err`. In the event that `historical_out` or `historical_err` aren't created yet, it'll throw the `FileNotFoundError` and a symlink will not be created.